### DISCUSSION TITLE
Sending a file between 2 Android phones using iperf

### DISCUSSION THREAD

**Discussion by fahad-ece** [esnet/iperf#1658] [Q&A]
Dear All,

I am seeking your help on an issue. I am trying to send a file using iperf from one android phone to another android phone. The phones are assigned with IP address of 10.45.0.4 and 10.45.0.5 respectively. Both of them are connected to the internet through IP forwarding (using IP 10.45.0.1). Unfortunately, I can not even ping one android phone from another android phone. I have tried all the recommended apps from Google Playstore which includes - pingtools, fing, he.net-Network Tools and few other tools as well but without any success. Any help/suggestion to solve this issue will be highly appreciated. Thank you.

**Comment by davidBar-On** [comment]
> The phones are assigned with IP address of 10.45.0.4 and 10.45.0.5 respectively. Both of them are connected to the internet through IP forwarding (using IP 10.45.0.1).

Can you add mode details?  If both phones have 10.45.0.x IP addresses, then I assume they are connected a local 10.45.0.x subnet, e.g through WiFi.  Is this the case?  If it is, why do you need the Port Forwarding (NAT) for pinging through the internet?  Why not pinging directly through the local subnet?

If the above is not the case, can you explain how these IP addresses are allocated to the phones?  What is the 10.45.0.1 that the internet traffic is supposed to be tunneled through and how it is connected to the internet? How the phones are connected to this 10.45.0.1 router?

**Reply by fahad-ece** [reply]
Dear David, @davidBar-On 

Thanks for your comment. Actually I am working on a lab where we need to create 5G SA testbed network which is based on software defined radio (srsRAN CU/DU and Open5GS as core). So, I am using two android mobile phones to get connected to the network. Using WireShark I was able to determine the IP address of the mobile phones which are like 10.45.0.x, 10.45.0.y. 

Port forwarding is required if we want internet connectivity otherwise it is not required. What I have understood the default gateway for the mobile phones are 10.45.0.1 but somehow it is not possible to **iperf** the mobiles phones. Infact I am trying to understand how can I send a file from one mobile phone to another through the network.  I was using the "Ping Tools" from app store which could not even ping the other phone. I don't know how to make it happen. Because no one tried it earlier.

It will be a great help for me if you have any further suggestions for me. I am sharing the link as well from where I got the idea. Thank you.

Link: [https://docs.srsran.com/projects/project/en/latest/tutorials/source/cotsUE/source/index.html](url)

**Comment by davidBar-On** [comment]
Hi @ff187,
I am not familiar with the srsRAN.  I briefly viewed the document you referred to but I still don't understand the configuration of your network.  See for example the [5G Lab Architecture diagram in this page](https://www.amantyatech.com/5g/5g-use-case-lab-setup).  If it is similar to your network configuration, then I assume that the **5G Core** implements the 10.45.0.x network, and the **Router**, which is the gateway to the internet, is assigned with IP address 10.45.0.1.

If the above diagram is similar to your network configuration, the 10.45.0.4/5 UEs (phones) do **not** need to go through the internet (i.e. through the Router) to connect to each other.  I therefore don't understand why the port forwarding through the router is needed for your ping/iperf tests.

Few more questions / suggestion to check:

- Are you able to connect to servers on the internet from the phones?  E.g view web pages or iperf to a server on the internet.
- Similar question regarding pinging to servers on the internet.  Especially the ICMP ping, e.g. using this option of PingTools.
- As you are using Wireshark, try to capture the packets sent by ping/iperf.  Analyzing these packets can give a valuable information why the tests don't run.  If you can it can help to run Wireshark on different point in the network, to see which packets are sent, which are received, etc.  E.g. run Wireshark for each phone and the router.  If you are able to capture the Wireshark files, you can attach them to this discussion.

**Reply by fahad-ece** [reply]
Dear @davidBar-On,

Thanks a lot for your valuable feedback. Like I have said- port forwarding is required to get the internet connectivity. Otherwise the phones get connected with the Router very easily. I can perform "Traceroute (for example Google)" but could not ping the mobile phones from one another. 

I am sharing the Wireshark Captured Packet for your consideration and feedback. Thank you.

[Link of Captured Packet](https://drive.google.com/file/d/1FV9VyiJS7uOs2Mdl6Y3U4NUfaYGIw2mx/view?usp=sharing)

**Comment by davidBar-On** [comment]
Hi @ff187 , the problem is probably in the IP routing rules that do allow routing back to the 5G code.  I am not familiar with Open5GS but the following discussion can help as it seem to be about the same issue: [https://github.com/open5gs/open5gs/discussions/1084](https://github.com/open5gs/open5gs/discussions/1084).

**Reply by fahad-ece** [reply]
Dear @davidBar-On ,
Thanks a lot for your valuable feedback. I will go through it and get back to you at the earliest. Thank you.
